pioneer/packages/joy-media/src/common/MediaPlayerWithResolver.tsx

```diff
@@ -24,56 +24,46 @@ function newCancelSource (): CancelTokenSource {
 
 function InnerComponent (props: Props) {
   const { contentId, api, discoveryProvider } = props;
+  const transport = useTransport();
 
   const [error, setError] = useState<Error>();
   const [resolvedAssetUrl, setResolvedAssetUrl] = useState<string>();
   const [contentType, setContentType] = useState<string>();
   const [cancelSource, setCancelSource] = useState<CancelTokenSource>(newCancelSource());
 
-  const getActiveStorageProviderIds = async (): Promise<StorageProviderId[]> => {
-    const nextId = await api.query.storageWorkingGroup.nextWorkerId() as StorageProviderId;
-    // This is chain specfic, but if next id is still 0, it means no workers have been added,
-    // so the workerById is empty
-    if (nextId.eq(0)) {
-      return [];
-    }
-
-    const workers = new MultipleLinkedMapEntry<StorageProviderId, Worker>(
-      StorageProviderId,
-      Worker,
-      await api.query.storageWorkingGroup.workerById()
-    );
-
-    return workers.linked_keys;
-  };
-
   const resolveAsset = async () => {
     setError(undefined);
     setCancelSource(newCancelSource());
 
-    const rids: DataObjectStorageRelationshipId[] = await api.query.dataObjectStorageRegistry.relationshipsByContentId(contentId) as any;
+    const rids = await api.query.dataObjectStorageRegistry.relationshipsByContentId<Vec<DataObjectStorageRelationshipId>>(contentId);
 
-    const allRelationships: Option<DataObjectStorageRelationship>[] = await Promise.all(rids.map((id) => api.query.dataObjectStorageRegistry.relationships(id))) as any;
+    const allRelationships = await Promise.all(
+      rids.map((id) =>
+        api.query.dataObjectStorageRegistry.relationships<Option<DataObjectStorageRelationship>>(id)
+      )
+    );
 
     // Providers that have signalled onchain that they have the asset
-    let readyProviders = allRelationships.filter(r => r.isSome).map(r => r.unwrap())
-      .filter(r => r.ready)
-      .map(r => r.storage_provider);
+    let readyProviders = allRelationships.filter((r) => r.isSome).map((r) => r.unwrap())
+      .filter((r) => r.ready)
+      .map((r) => r.storage_provider);
 
     // runtime doesn't currently guarantee unique set
-    readyProviders = _.uniqBy(readyProviders, provider => provider.toString());
+    readyProviders = _.uniqBy(readyProviders, (provider) => provider.toString());
 
     if (!readyProviders.length) {
       setError(new Error('No Storage Providers found storing this content'));
+
       return;
     }
 
+    const activeProviders = (await transport.workingGroups.allWorkers('Storage')).map(([id]) => id);
```

Comment:
In this implementation we need to filter out the Storage Group Lead id, so we get only the storage providers.

Reply:
After raising this in chat, we can ignore this comment.

ref: https://github.com/Joystream/joystream/issues/1211